pkg/runtime/interfaces.go

```diff
@@ -103,18 +107,11 @@ type ObjectCreater interface {
 	New(version, kind string) (out Object, err error)
 }
 
-// ObjectDecoder is a convenience interface for identifying serialized versions of objects
-// and transforming them into Objects. It intentionally overlaps with ObjectTyper and
-// Decoder for use in decode only paths.
-type ObjectDecoder interface {
-	Decoder
-	// DataVersionAndKind returns the version and kind of the provided data, or an error
-	// if another problem is detected. In many cases this method can be as expensive to
-	// invoke as the Decode method.
-	DataVersionAndKind([]byte) (version, kind string, err error)
-	// Recognizes returns true if the scheme is able to handle the provided version and kind
-	// of an object.
-	Recognizes(version, kind string) bool
+// ObjectCopier duplicates an object.
+type ObjectCopier interface {
+	// Copy returns an exact copy of the provided Object, or an error if the
```

Comment:
Thanks - I will be touching this file anyway very soon - so I will update it then.